### ISSUE TITLE
Virtual GPhoto device driver for testing client applications

### ISSUE THREAD

**Issue by yecril71pl** [gphoto/libgphoto2#408]
Fill in fields below. If you don't know how, leave it free.

**Name of the camera*
Virtual GPhoto

**USB IDs**
TBD

**camera summary output**
TBD

**camera configuration output**
TBD

**test capture**
A predefined image

**Comment by msmeissn** [comment]
libgphoto2 has a virtual PTP camera, configure with --enable-vusb  and no other libusb libraries.
a bit tricky to handle, but should work.

**Comment by yecril71pl** [comment]
I cannot find it documented.  Not in INSTALL and not in [configure help](https://build.opensuse.org/package/live_build_log/home:yecril71pl:branches:graphics/libgphoto2-virtual/openSUSE_Leap_15.0/x86_64).

**Comment by tkircher** [comment]
I'll add the documentation here since it doesn't seem to exist anywhere else.

In order for the virtual camera device to work, like @msmeissn suggested above, you must enable vusb and explicitly disable all other usb libraries:

`./configure --enable-vusb --without-libusb --with-libusb-1.0=no`

It is also important to make sure you don't have any other versions of the library installed. So, on a clean system without libgphoto2 or gphoto2 installed, build libgphoto2, then install it, and build gphoto2 and install it. The device doesn't come up as `vusb:`, rather
```
$ gphoto2 --list-ports
...
usb:001,001                      Universal Serial Bus
```

```
$ gphoto2 --auto-detect
Model                          Port                                            
----------------------------------------------------------
Nikon DSC D750                 usb:001,001     
```

```
$ gphoto2 -a
Abilities for camera             : Nikon DSC D750                              
Serial port support              : no
USB support                      : yes
Capture choices                  :
                                 : Image
                                 : Preview
                                 : Trigger Capture
Configuration support            : yes
Delete selected files on camera  : yes
Delete all files on camera       : no
File preview (thumbnail) support : yes
File upload support              : no
```

**Comment by pmnlla** [comment]
This is done with the configuration script at build time, right?

**Comment by yecril71pl** [comment]
I have [reconfigured the library](https://build.opensuse.org/build/home:yecril71pl:branches:graphics/openSUSE_Tumbleweed/x86_64/libgphoto2/_log) and it [did not work](https://build.opensuse.org/package/live_build_log/home:yecril71pl:branches:graphics/gphoto2-vusb-test/openSUSE_Tumbleweed/x86_64) :disappointed: 

> [   52s] + gphoto2 -a
> [   52s] Could not detect any camera

**Comment by tkircher** [comment]
> I have [reconfigured the library](https://build.opensuse.org/build/home:yecril71pl:branches:graphics/openSUSE_Tumbleweed/x86_64/libgphoto2/_log) and it [did not work](https://build.opensuse.org/package/live_build_log/home:yecril71pl:branches:graphics/gphoto2-vusb-test/openSUSE_Tumbleweed/x86_64) 😞
> 
> > [   52s] + gphoto2 -a
> > [   52s] Could not detect any camera

Looks like you installed libgphoto2 from some package source, rather than building it:

> [   42s] libgphoto2_port12-2.5.30-250.1        ########################################

**Comment by yecril71pl** [comment]
> Looks like you installed libgphoto2 from some package source, rather than building it:
> 
> > [   42s] libgphoto2_port12-2.5.30-250.1        ########################################

I have [packaged it](https://build.opensuse.org/build/home:yecril71pl:branches:graphics/openSUSE_Tumbleweed/x86_64/libgphoto2/_log) myself.

**Comment by msmeissn** [comment]
i sent you a fix, osc rq accept -m ok 1082175

(basically FUZZ_PTP was undefined, so no camera was reported ... #define FUZZ_PTP makes it report the D750 id)

**Comment by msmeissn** [comment]
i sent you a fix, osc rq accept -m ok 1082175

(basically FUZZ_PTP was undefined, so no camera was reported ... #define FUZZ_PTP makes it report the D750 id)
